src/activation_token/mod.rs

```diff
@@ -0,0 +1,119 @@
+#[cfg(all(feature = "gtk3", feature = "wayland"))]
+mod gtk3;
+#[cfg(all(feature = "gtk3", feature = "wayland"))]
+pub use self::gtk3::Gtk3ActivationToken;
+
+#[cfg(feature = "gtk4_wayland")]
+mod gtk4;
+#[cfg(feature = "gtk4_wayland")]
+pub use self::gtk4::Gtk4ActivationToken;
+
+#[cfg(any(feature = "wayland"))]
+mod wayland;
+#[cfg(feature = "wayland")]
+pub use wayland::WaylandActivationToken;
+
+use serde::{ser::Serializer, Serialize};
+use zbus::zvariant::Type;
+
+// TODO
+/// See https://wayland.app/protocols/xdg-activation-v1
+#[derive(Type)]
+#[zvariant(signature = "s")]
+#[derive(Debug)]
+pub enum ActivationToken {
+    #[cfg(feature = "wayland")]
+    #[doc(hidden)]
+    Wayland(WaylandActivationToken),
+    #[cfg(feature = "gtk4_wayland")]
+    #[doc(hidden)]
+    Gtk4(Gtk4ActivationToken),
+    #[cfg(all(feature = "gtk3", feature = "wayland"))]
+    #[doc(hidden)]
+    Gtk3(Gtk3ActivationToken),
+    #[doc(hidden)]
+    Raw(String),
+    #[doc(hidden)]
+    None,
+}
+
+impl Serialize for ActivationToken {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(self.as_str())
+    }
+}
+
+impl Default for ActivationToken {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
+impl ActivationToken {
+    #[cfg(feature = "wayland")]
+    /// Create an instance of [`ActivationToken`] from a Wayland surface and the
+    /// application's id.
+    pub async fn from_wayland_surface(
+        app_id: &str,
```

Comment:
We already have plenty of places where we use the `app_id` so maybe we should have a type for it that wraps a `String` or something like that.

Reply:
I think its way more convenient for the user to just write their app_id in a &str, ideally it would be preferable to somehow get it from the wayland toplevel or gtkwindow..

Reply:
Sure, you could make those functions take a `impl Into<AppID>` and `impl From<&str> for AppID` but as it is something that can be done in a separate PR, I will handle it then you can just rebase.